feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt

```diff
@@ -61,5 +49,76 @@ fun Product.toStoreProduct(marketplace: String): StoreProduct {
     )
 }
 
+internal fun String.extractPrice(
+    currency: Currency
+): Price {
+    debugLog("Received price is $this. Extracting currency and amount.")
+
+    val (priceNumeric, currencySymbol) = this.parsePriceAndCurrencySymbolUsingRegex() ?: 0.0f to currency.symbol
+
+    debugLog("Extracted price is: $priceNumeric. Currency symbol is $currencySymbol")
+
+    var foundCurrencyCode: String? = null
+    if (currencySymbol == currency.symbol) {
+        debugLog("Currency symbol matches Locale's.")
+        foundCurrencyCode = currency.currencyCode
+    } else {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            // We've seen cases the price being as "US$", but the marketplace currency code being "INR"
+            debugLog("Currency symbol is not the same as Locale's. Iterating over all available currencies to find " +
+                "the currency code.")
+            foundCurrencyCode =
+                Currency.getAvailableCurrencies()
+                    .firstOrNull { it.symbol == currencySymbol }?.currencyCode
+        }
+    }
+
+    val currencyCode = foundCurrencyCode ?: run {
+        debugLog("Couldn't determine currencyCode. Setting currencyCode to symbol sent by Amazon")
+        currencySymbol
+    }
+
+    debugLog("Currency code is $currencyCode")
+
+    return Price(
+        currencyCode,
+        priceAmountMicros = priceNumeric.times(MICROS_MULTIPLIER).toLong()
+    )
+}
+
+internal data class Price(
+    val currencyCode: String,
+    val priceAmountMicros: Long
+)
+
+// Explanations about the regexp:
+// \\d+: match the first(s) number(s)
+// [\\.,\\s]: match a "separator": a dot, comma or space
+// \\d+ (the second one): match the number(s) after the separator.
+// The lasts two are englobed in []*, as they can be repeated 0 or n times.
+private val pattern: Pattern = Pattern.compile("(\\d+[[\\.,\\s]\\d+]*)")
+
+internal fun String.parsePriceAndCurrencySymbolUsingRegex(): Pair<Float, String>? {
```

Comment:
since this is pretty complex, i wonder if we should break into smaller unit-testable pieces